apps/client/src/components/conditionSelector/ConditionSelector.vue

```diff
@@ -0,0 +1,317 @@
+                </div>
+                <div class="items-center justify-around row">
+                    <template
+                        v-for="(labelX, idx) in xLabels"
+                        :key="idx"
+                    >
+                        <div
+                            @mouseover="
+                                () => handleLabelMouseOver('x-axis', idx)
+                            "
+                            @mouseleave="() => handleLabelMouseLeave()"
+                            @click="() => conditionSelector.clickConditionChartColumn(idx)"
+
+                            :class="`row justify-center align-center flex x-label ${hoveredAll ? 'hovered' : ''}`"
+                        >
+                            <div :style="width">
+                                {{ labelX }}
+                            </div>
+                        </div>
+                    </template>
+                </div>
+            </div>
+            <div class="items-start justify-center flex">
+                <svg width="50" style="position:relative; overflow:visible;" height="50" xmlns="http://www.w3.org/2000/svg">
+                    <line x1="0" y1="20" x2="15" y2="20" stroke="black" stroke-width="3" />
+                    <text x="20" y="23" font-family="Arial" font-size="10" fill="black">Selected</text>
+                    <line x1="0" y1="35" x2="15" y2="35" stroke="black" stroke-width="2" />
+                    <text x="20" y="42" font-family="Arial" font-size="10" fill="black">Unselected</text>
+                    <rect fill="#cccccc" x="0" width="15" y="36" height="8"/>
+                </svg>
+            </div>
+            <div class="items-center justify-center flex x-tag">
+                <ConditionSelectorDropDown axis="x-axis" />
+            </div>
+        </div>
+    </div>
+</template>
+
+<style scoped lang="scss">
+
+$border: 1px solid #9ca3af;
+.condition-selector-container {
+    grid-template-columns: 60px 1fr;
+    grid-template-rows: 1fr 60px;
+    font-size: 0.8rem;
+    display: grid;
+    width: 100%;
+    height: 100%;
+    & > div:not(.condition-charts-container) {
+        width: 100%;
+        height: 100%;
+    }
+    .y-tag {
+        font-size: 1.2rem;
+        div {
+            transform: rotate(-90deg);
+        }
+    }
+    .x-tag {
+        font-size: 1.2rem;
+    }
+
+    .condition-charts-container {
+        grid-template-columns: 60px 1fr;
+        grid-template-rows: 1fr 60px;
+        display: grid;
+        width: 100%;
+        height: 100%;
+        & > div {
+            width: 100%;
+            height: 100%;
+        }
+        .y-label {
+            width: 100%;
+            flex: 1;
+        }
+        .x-label {
+            height: 100%;
+        }
+        .y-label,
+        .x-label {
+            cursor: pointer;
+            text-align: center;
+            border-radius: 2px;
+            &.hovered,
+            &:hover {
+                border: $border;
```

Comment:
Actually never knew this. This is similar to using border with box-sizing:border-box (which makes it so that borders and padding are accounted for when setting width and height)